messaging/src/main/java/org/axonframework/eventhandling/scheduling/dbscheduler/SimpleDbSchedulerEventSchedulerSupplier.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.scheduling.dbscheduler;
+
+import org.axonframework.deadline.dbscheduler.DbSchedulerDeadlineManager;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+
+/**
+ * Default supplier for an {@link DbSchedulerEventScheduler}. This makes it easier to use in context without more
+ * advanced ways of dependency injection. It can be passed to the tasks in the {@link DbSchedulerDeadlineManager} to
+ * create the {@link com.github.kagkarlsson.scheduler.Scheduler} before the {@link DbSchedulerEventScheduler} is
+ * created. After creating it should be set on this {@link SimpleDbSchedulerEventSchedulerSupplier}.
+ *
+ * @author Gerard Klijs
+ * @since 4.9.0
+ */
+public class SimpleDbSchedulerEventSchedulerSupplier implements Supplier<DbSchedulerEventScheduler> {
```

Comment:
Why not just call it the `DbSchedulerEventSchedulerSupplier`? The class name now slightly confuses me on the point that I expect a `SimpleDbSchedulerEventScheduler` to be supplied (although that doesn't exist), while it actually supplies a `DbSchedulerEventScheduler`.

Reply:
I was struggling with the name. Removing `Simple` seems fine indeed.